pages/index.tsx

```diff
@@ -28,14 +29,46 @@ import {
   Synth,
 } from "../utils/umaApi";
 
-import { getDefillamaTvl, getDefillamaPercentChange } from "../utils/defillama";
+import {
+  getDefillamaTvl,
+  getDefillamaPercentChange,
+  getDefillamaStats,
+} from "../utils/defillama";
+import { ethers } from "ethers";
+
+async function attachDefillamaStats(cmsSynth: ContentfulSynth) {
+  let data;
+
+  try {
+    const { tvl, tvl24hChange } = await getDefillamaStats(cmsSynth.externalApi);
+    data = {
+      ...cmsSynth,
+      tvl: ethers.utils.parseEther(tvl.toFixed(0)).toString(),
+      tvl24hChange,
+    };
+  } catch (error) {
+    return error;
+  }
+
+  return data;
+}
+
+function fetchCompleteSynthByApi(cmsSynth: ContentfulSynth) {
+  if (cmsSynth.externalApi) {
+    return attachDefillamaStats(cmsSynth);
+  } else {
+    return fetchCompleteSynth(cmsSynth);
+  }
+}
 
 export const getStaticProps: GetStaticProps = async () => {
   const queryClient = new QueryClient();
   const cmsSynths = await contentfulClient.getAllSynths();
 
   await queryClient.prefetchQuery("all synths", async () =>
-    (await Promise.all(cmsSynths.map(fetchCompleteSynth))).filter(errorFilter)
+    (
+      await Promise.all(cmsSynths.map(fetchCompleteSynthByApi))
```

Comment:
does this imply that if a contract has `.externalUrl` it also needs to have `externalApi` defined as well or it will be hidden?  i guess im worried that theres some case where contentful contracts get hidden due to erroring in the api when it shouldnt matter

Reply:
We can discuss this edge case, yes. What should happen if defillama api fails. Display `tvl` and `24hChange` as `N/A`?

Reply:
The existing implementation hides the synths if the calls to our api fails

Reply:
yea i would say show anyways in a degraded state, but we can revisit this